app/src/main/java/space/taran/arknavigator/mvp/presenter/adapter/tagsselector/TagsSelectorPresenter.kt

```diff
@@ -116,6 +136,60 @@ class TagsSelectorPresenter(
         viewState.drawTags()
     }
 
+    fun onBackClick(): Boolean {
+        verifyActions()
+
+        if (actions.isEmpty())
+            return false
+
+        when(val action = actions.last()) {
+            is Include -> {
+                included.remove(action.tag!!)
+            }
+            is Exclude -> {
+                excluded.remove(action.tag!!)
+            }
+            is UncheckIncluded -> {
+                included.add(action.tag!!)
+            }
+            is UncheckExcluded -> {
+                excluded.add(action.tag!!)
+            }
+            is UncheckAndExclude -> {
+                excluded.remove(action.tag!!)
+                included.add(action.tag)
+            }
+            is Clear -> {
+                included = action.included.toMutableSet()
+                excluded = action.excluded.toMutableSet()
+            }
+        }
+
+        actions.removeLast()
+
+        calculateTagsAndSelection()
+
+        return true
+    }
+
+    private fun verifyActions() {
+        if (storage == null || index == null)
+            return
```

Comment:
`storage` and `index` are `null`'s while indexing is in progress and the user can tap "back"
But we handle this case by checking `actions.isEmpty()`

Reply:
If we handle this case in other place (`onBackClick()` IIUC), then this `return` is confusing since we never reach it in fact. I would replace it with `AssertionError` then to catch a bug if we are wrong. Technically, we need to put it only for debug mode, but at this stage of project we don't really have production :)